plugins/uefi/fu-uefi-common.c

```diff
@@ -97,11 +97,19 @@ fu_uefi_get_built_app_path (GError **error)
 				       suffix,
 				       extension);
 	if (!g_file_test (source_path, G_FILE_TEST_EXISTS)) {
-		g_set_error (error,
-			     G_IO_ERROR,
-			     G_IO_ERROR_NOT_FOUND,
-			     "%s cannot be found",
-			     source_path);
+		if (fu_uefi_secure_boot_enabled ()) {
+			g_set_error (error,
+				G_IO_ERROR,
+				G_IO_ERROR_NOT_FOUND,
+				"%s is required for secure boot",
+				source_path);
```

Comment:
Isn't this in the wrong place? `fu_uefi_get_built_app_path()` is a helper, rather than anything that knows the purpose of the call. i.e. I think it needs to be a g_prefix_error in the caller.

Reply:
Adjusted.